app/frontend/src/features/auth/AuthProvider.tsx

```diff
@@ -13,6 +16,19 @@ function useAppContext<T>(context: Context<T | null>) {
 
 export default function AuthProvider({ children }: { children: ReactChild }) {
   const store = useAuthStore();
+
+  const push = useHistory().push;
+
+  useEffect(
+    () =>
+      setUnauthenticatedErrorHandler(() => {
+        store.authActions.logout();
+        store.authActions.authError("You were logged out.");
+        push(loginRoute);
+      }),
+    [push, store.authActions]
+  );
```

Comment:
```suggestion
  useEffect(
    () => {
      setUnauthenticatedErrorHandler(() => {
        store.authActions.logout();
        store.authActions.authError("You were logged out.");
        push(loginRoute);
      });

      return () => {
        setUnauthenticatedErrorHandler(() => {});
      };
    },
    [push, store.authActions]
  );
```
Might be best to clean up the effect by unsetting the handler to a noop function to reduce any possibility of a memory leak if you are sticking with this solution...